Vwitter/SpotifyAPI/SpotifyAPIManager.m

```diff
@@ -0,0 +1,212 @@
+        NSLog(@"authorized spotify session");
+    } else {
+        // Use this on iOS versions < 11 to use SFSafariViewController
+        NSLog(@"iOS version too old");
+    }
+    
+    SPTConfiguration *configuration =
+        [[SPTConfiguration alloc] initWithClientID:SpotifyClientID redirectURL:[NSURL URLWithString:SpotifyRedirectURLString]];
+
+    self.appRemote = [[SPTAppRemote alloc] initWithConfiguration:configuration logLevel:SPTAppRemoteLogLevelDebug];
+}
+
+- (void)authorizeSpotify {
+    [self setupAppRemote];
+    [self.appRemote authorizeAndPlayURI:@"spotify:artist:2YZyLoL8N0Wb9xBt1NhZWg"];
+}
+
+- (void)playTrack:(NSString *)trackUri {
+    if (!self.hasValidSpotifyAuthorization) {
+        [self authorizeSpotify];
+    }
+    
+    if (!self.appRemote.playerAPI) {
+        [self.appRemote connect];
+    }
+    
+    [self.appRemote.playerAPI play:trackUri callback:^(id  _Nullable result, NSError * _Nullable error) {
+        if (!error) {
+            NSLog(@"track playing");
+        }
+        else {
+            NSLog(@"cell track failed to play %@", error.localizedDescription);
+        }
+    }];
+}
+
+- (void)pause {
+    [self.appRemote.playerAPI pause:^(id  _Nullable result, NSError * _Nullable error) {
+        if (!error) {
+            NSLog(@"track paused");
+
+        }
+        else {
+            NSLog(@"track not paused, %@", error.localizedDescription);
+        }
+    }];
+    
+}
+
+- (void)getTracks:(NSString *)searchString withCompletion:(SpotifyTrackCompletion)completion {
+    if (!self.hasValidSpotifyAuthorization) {
+        [self authorizeSpotify];
+    }
+    
+    NSString *bearerToken = [[NSString alloc] initWithFormat:@"Bearer %@", self.appRemote.connectionParameters.accessToken];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    
+    NSString *convertedSearchString = [searchString stringByReplacingOccurrencesOfString:@" " withString:@"%20"];
+    
+    NSString *urlString = [NSString stringWithFormat:@"%@/%@/%@", @"https://api.spotify.com/v1/search?q=", convertedSearchString, @"&type=track"];
+    NSURL *url = [NSURL URLWithString:urlString];
+
+    [request setValue:bearerToken forHTTPHeaderField:@"Authorization"];
+    [request setURL:url];
+
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+      [[session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (!error) {
+            NSMutableArray *spotifyTracksArray = [[NSMutableArray alloc] init];
+            
+            NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+            // JSON with song is here
+            NSLog(@"JSON: %@", json);
+            NSArray *tracksArray = json[@"tracks"][@"items"];
+            NSArray *castedTracksArray = CAST_TO_CLASS_OR_NIL(tracksArray, NSArray);
```

Comment:
nit: this can be simplified to one line and you can remove `tracksArray`

NSArray *castedTracksArray = CAST_TO_CLASS_OR_NIL(json[@"tracks"][@"items"], NSArray);